doc/source/whatsnew/v1.3.0.rst

```diff
@@ -142,7 +142,7 @@ Other API changes
 
 Deprecations
 ~~~~~~~~~~~~
-
+- Deprecated allowing subclass-specific keyword arguments in the :class:`Index` constructor, use the specific subclass directly instead (:issue:`22315`)
```

Comment:
can you add all of the issues that you listed in the top of the PR here.

Reply:
its a little ambiguous, might want to not close until the deprecation is enforced?

Reply:
i would list them and we can close them with this PR. its not worth keeping them around now that they are deprecated. if you want to open a dedicated issues that lists the things that we want to check once this deprecation is enforced ok by me. (e.g. simply linking to those 4 issues is fine as well).

Reply:
updated + greenish